app/src/main/java/com/duckduckgo/app/browser/trafficquality/remote/AndroidFeaturesHeaderProvider.kt

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.trafficquality.remote
+
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.autoconsent.api.Autoconsent
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.app.tracking.AppTrackingProtection
+import com.duckduckgo.networkprotection.api.NetworkProtectionState
+import com.duckduckgo.privacy.config.api.Gpc
+import com.squareup.anvil.annotations.ContributesBinding
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
+
+interface AndroidFeaturesHeaderProvider {
+    fun provide(): String?
+}
+
+@ContributesBinding(AppScope::class)
+class RealAndroidFeaturesHeaderProvider @Inject constructor(
+    private val appBuildConfig: AppBuildConfig,
+    private val featuresRequestHeaderStore: FeaturesRequestHeaderStore,
+    private val autoconsent: Autoconsent,
+    private val gpc: Gpc,
+    private val appTrackingProtection: AppTrackingProtection,
+    private val networkProtectionState: NetworkProtectionState,
+) : AndroidFeaturesHeaderProvider {
+    override fun provide(): String? {
+        val config = featuresRequestHeaderStore.getConfig()
+        if (config.isNotEmpty()) {
+            val versionConfig = config.find { it.appVersion == appBuildConfig.versionCode }
+            if (versionConfig != null) {
+                if (shouldLogValue(versionConfig)) {
+                    return mapFeatures(versionConfig)
+                }
+            } else {
+                return null
+            }
+        } else {
+            return null
+        }
+
+        return null
+    }
+
+    private fun shouldLogValue(versionConfig: TrafficQualityAppVersion): Boolean {
+        val appBuildDateMillis = appBuildConfig.buildDateTimeMillis
+        if (appBuildDateMillis == 0L) {
+            return false
+        }
+
+        val appBuildDate = LocalDateTime.ofEpochSecond(appBuildDateMillis / 1000, 0, ZoneOffset.UTC)
+        val now = LocalDateTime.now(ZoneOffset.UTC)
+
+        val daysSinceBuild = ChronoUnit.DAYS.between(appBuildDate, now)
+        val daysUntilLoggingStarts = versionConfig.daysUntilLoggingStarts
+        val daysForAppVersionLogging = versionConfig.daysUntilLoggingStarts + versionConfig.daysLogging
+
+        return daysSinceBuild in daysUntilLoggingStarts..daysForAppVersionLogging
+    }
+
+    private fun mapFeatures(versionConfig: TrafficQualityAppVersion): String? {
+        return runBlocking {
```

Comment:
`runBlocking` will likely cause ANRs here. Would it make sense to cache these toggles on init so that they can be accessed synchronously? 

Here’s an example: https://app.asana.com/0/0/1208676143593571/1208701434848757/f